src/VerwerkingenLambdaFunction/handler.py

```diff
@@ -0,0 +1,224 @@
+    }
+
+def get_verwerkingsacties_actieid(event, table):
+    ######################################
+    ## GET /verwerkingsacties/{actieId} ##
+    ######################################
+    response = table.query(
+            KeyConditionExpression=Key('actieId').eq(event['queryStringParameters']['actieId'])
+    )
+    
+    # Check if requested record is found. If not, the list of items is empty (0).
+    if ( len(response.get('Items')) == 0 ):
+        return {
+            'statusCode': 400,
+            'body': 'Record not found',
+            'headers': { "Content-Type": "text/plain" },
+        }
+    else:
+        return {
+            'statusCode': 200,
+            'body': json.dumps(response.get('Items')[0]),
+            'headers': { "Content-Type": "application/json" },
+        }
+
+def put_verwerkingsacties_actieid(event, table):
+    ######################################
+    ## PUT /verwerkingsacties/{actieId} ##
+    ######################################
+    requestJSON = json.loads(event['body'])
+    actieId = event['queryStringParameters']['actieId']
+    item = filled_item(requestJSON, actieId, "2024-04-05T14:36:42+01:00")
+
+    table.put_item(
+        Item=item
+    )
+
+    return {
+        'statusCode': 200,
+        'body': json.dumps(item),
+        'headers': { "Content-Type": "application/json" }
+    }
+
+def delete_verwerkingsacties_actieid(event, table):
+    #########################################
+    ## DELETE /verwerkingsacties/{actieId} ##
+    #########################################
+    response = table.delete_item(
+        Key={
+            'actieId': event['queryStringParameters']['actieId'],
+        }
+    )
+
+    return {
+        'statusCode': 200,
+        'body': json.dumps(response),
+        'headers': { "Content-Type": "application/json" }
+    }
+
+def store_item_in_s3(item_json, bucket):
+    """Store (backup) verwerking item in S3 Backup Bucket
+    """
+    path = datetime.now().isoformat(timespec='seconds') + "_" + json.loads(item_json)['actieId']
+    data = bytes(item_json.encode('UTF-8'))
+    bucket.put_object(
+        ContentType='application/json',
+        Key=path,
+        Body=data,
+    )
+
+def handle_request(event, table, bucket):
+    """Receives the event object and routes it to the correct function
+    """
+    params = parse_event(event)
+    if(params['method'] == 'GET' and params['resource'] == '/verwerkingsacties'):
+        return get_verwerkings_acties(event, table)
+
+    if(params['method'] == 'POST' and params['resource'] == '/verwerkingsacties'):
+        result = post_verwerkings_acties(event, table)
+        store_item_in_s3(result['body'], bucket)
+        return result
+
+    if(params['method'] == 'PATCH' and params['resource'] =='/verwerkingsacties'):
+        return patch_verwerkings_acties(event, table)
+
+    if(params['method'] == 'GET' and params['resource'] == '/verwerkingsacties/{actieId}'):
```

Comment:
Bij deze en de PUT lees ik de API-specificatie zo dat waar {actieId} staat de daadwerkelijke actieId in de url meegegeven wordt. De query-parameter actieId is niet nodig zo te zien. Zie https://petstore.swagger.io/?url=https://raw.githubusercontent.com/VNG-Realisatie/gemma-verwerkingenlogging/master/docs/api-write/oas-specification/logging-verwerkingen-api/openapi.yaml#/verwerkingsacties/verwerkingsactie_read